sdk/core/azure_core_amqp/src/fe2o3/cbs.rs

```diff
@@ -69,12 +74,22 @@ impl AmqpClaimsBasedSecurityApis for Fe2o3ClaimsBasedSecurity {
                     .to_offset(time::UtcOffset::UTC)
                     .unix_timestamp()
                     .checked_mul(1_000)
-                    .unwrap(),
+                    .ok_or_else(|| {
+                        azure_core::Error::message(
+                            azure_core::error::ErrorKind::Other,
```

Comment:
Not necessarily for this PR, but I think this'll happen enough where I'm wondering if we should add a new `ErrorKind` or - at least in this case - if `DataConversion` (for which you can still pass a custom message or `Error` is more appropriate.

Reply:
I'm currently rethinking my entire "how I handle errors" philosophy for both AMQP and Eventhubs based on comments you've made in the past.

I'd love to chat a bit about brainstorming how we reconcile service specific errors with azure_core::Error - which doesn't really have a mechanism to express specific errors.

This shows up with Eventhubs a bunch, where there are a number of state related errors that can occur (for instance, if you attempt to get properties on a `ProducerClient` before you call `open` on the client) but they end up getting glommed into `azure_core::error::ErrorKind::Other` because there isn't a better `ErrorKind` enumerator that represents service specific errors.